File: numeric/euclidgcd/euclidgcd.swift
import Foundation

func euclidgcd(m_in: Int, n_in: Int) -> Int {
    var m = m_in
    var n = n_in
    var r = 0
    while (n != 0) {
        r = m % n
        m = n
        n = r
    }
    return m
}

var m = 15
var n = 10

if (CommandLine.argc >= 3) {
    m = Int(CommandLine.arguments[1]) ?? 15
    n = Int(CommandLine.arguments[2]) ?? 10
}

let gcd = euclidgcd(m_in: m, n_in: n)

print("\(m) \(n)")
print("gcd: \(gcd)")

File: numeric/euclidgcd/euclidgcd.kt

fun euclidgcd(m_in: Int, n_in: Int): Int {
    var m = m_in
    var n = n_in
    var r = 0
    while (n != 0) {
        r = m % n
        m = n
        n = r
    } 
    return m
}

fun main(args: Array<String>) {
    val m = if (args.size >= 2) args[0].toInt() else 15
    val n = if (args.size >= 2) args[1].toInt() else 10
    val gcd = euclidgcd(m, n)

    println("$m $n")
    println("gcd: $gcd")
}
